gateways/c/fjage.c

```diff
@@ -256,7 +256,8 @@ static fjage_msg_t mqueue_get(fjage_gw_t gw, const char* clazz, const char* id)
   if (gw == NULL) return NULL;
   _fjage_gw_t* fgw = gw;
   fjage_msg_t msg = NULL;
-  for (int i = fgw->mqueue_tail; i != fgw->mqueue_head && msg == NULL; i++) {
+  for (int i = fgw->mqueue_tail; i != fgw->mqueue_head && msg == NULL;) {
+    i = (i + 1) % QUEUE_LEN;
```

Comment:
This should really go inside the `for`. It's not equivalent to put it here, as here it'll increment at the start of the loop rather than at the end.

Reply:
Yikes. I get you now.